juju/application.py

```diff
@@ -207,10 +208,12 @@ async def add_unit(self, count=1, to=None, attach_storage=[]):
             attach_storage=attach_storage,
         )
 
-        return await jasyncio.gather(*[
-            jasyncio.ensure_future(self.model._wait_for_new("unit", unit_id))
-            for unit_id in result.units
-        ])
+        return await asyncio.gather(
+            *[
+                asyncio.ensure_future(self.model._wait_for_new("unit", unit_id))
+                for unit_id in result.units
+            ]
+        )
```

Comment:
asyncio.gather already converts coroutines to tasks, there's no need to call .ensure_future.

let's simplify this.